Sources/MapboxDirections/VisualInstructionComponent.swift

```diff
@@ -171,6 +172,64 @@ public extension VisualInstruction.Component {
     }
 }
 
+/**
+ A mapbox shield representation of a visual instruction component.
+ */
+public struct MapboxShield: Equatable, Codable {
+    /**
+     Initializes a mapbox shield with the given name, text color, and display ref.
+     */
+    public init(baseURL: URL, name: String, textColor: String, displayRef: String) {
+        self.baseURL = baseURL
+        self.name = name
+        self.textColor = textColor
+        self.displayRef = displayRef
+    }
+    
+    /**
+     Base URL to query the styles endpoint.
+     */
+    public let baseURL: URL
+
+    /**
+     String indicating the name of the route shield.
+     */
+    public let name: String
+    
+    /**
+     String indicating the color of the text to be rendered on the route shield.
+     */
+    public let textColor: String
+    
+    /**
+     String indicating the display ref.
+     */
+    public let displayRef: String
```

Comment:
“Ref” is OpenStreetMap jargon that most developers would be unlikely to recognize. The Directions API uses this term for consistency with the `ref` property on the Step object, but this library uses the more intuitive term “reference code” with `code` as shorthand:

https://github.com/mapbox/mapbox-directions-swift/blob/0f20e9a3d5928333eb311dc3a6f6ae363a05a8a3/Sources/MapboxDirections/RouteStep.swift#L812-L819

Even so, what this property contains differs from the raw reference code in the Step object. Typically, a raw reference code would include an alphabetic network code, such as `US` for a U.S. Route, but many route networks omit that code from the shield, as in <img src="https://upload.wikimedia.org/wikipedia/commons/7/79/US_101.svg" height="24" align="top" alt="101">, which bears just the route number “101” instead of “US 101”. For this reason, the Directions API appends “display” to the property name, but that would have the effect of making it sound like a method in Swift. Instead, perhaps we could name this property `number`, as in route number, or `contents` if the possibility of an alphanumeric number would lead to confusion.

Reply:
`text` would be most appropriate here since there is the possibility of alphanumeric route numbering (Australia and the UK, among others, use this system).